test/tests/parallel/set3/process-check-restart.bats

```diff
@@ -0,0 +1,188 @@
+    --env ENABLE_FETCHMAIL=1
+    --env FETCHMAIL_PARALLEL=1
+    --env ENABLE_POSTGREY=1
+    --env ENABLE_SASLAUTHD=1
+    --env ENABLE_SRS=1
+    --env SMTP_ONLY=0
+    # Required workaround for some environments when using ENABLE_SRS=1:
+    # PR 2730: https://github.com/docker-mailserver/docker-mailserver/commit/672e9cf19a3bb1da309e8cea6ee728e58f905366
+    --ulimit "nofile=$(ulimit -Sn):$(ulimit -Hn)"
+  )
+  init_with_defaults
+  mv "${TEST_TMP_CONFIG}/fetchmail/fetchmail.cf" "${TEST_TMP_CONFIG}/fetchmail.cf"
+  # Average time: 6 seconds
+  common_container_setup 'CONTAINER_ARGS_ENV_CUSTOM'
+
+  local ENABLED_PROCESS_LIST=(
+    "${CORE_PROCESS_LIST[@]}"
+    "${ENV_PROCESS_LIST[@]}"
+  )
+
+  for PROCESS in "${ENABLED_PROCESS_LIST[@]}"
+  do
+    _should_restart_when_killed "${PROCESS}"
+  done
+
+  # By this point the fetchmail processes have been verified to exist and restart,
+  # For FETCHMAIL_PARALLEL=1 coverage, match full commandline for COUNTER values:
+  pgrep --full 'fetchmail-1.rc'
+  assert_success
+  pgrep --full 'fetchmail-2.rc'
+  assert_success
+}
+
+# Split into separate test case for the benefit of minimizing CPU + RAM overhead of clamd.
+# NOTE: Does not reduce test time of previous test case. Adds 10 seconds to test time.
+@test "(enabled ENV) should restart clamd when killed" {
+  export CONTAINER_NAME=${CONTAINER3_NAME}
+  local CONTAINER_ARGS_ENV_CUSTOM=(
+    --env ENABLE_CLAMAV=1
+  )
+  init_with_defaults
+  common_container_setup 'CONTAINER_ARGS_ENV_CUSTOM'
+
+  _should_restart_when_killed 'clamd'
+}
+
+function _should_restart_when_killed() {
+  local PROCESS=${1}
+  local MIN_PROCESS_AGE=4
+
+  # Wait until process has been running for at least MIN_PROCESS_AGE:
+  # (this allows us to more confidently check the process was restarted)
+  run_until_success_or_timeout 30 _check_if_process_is_running "${PROCESS}" "${MIN_PROCESS_AGE}"
+  # NOTE: refute_output doesn't have output to compare to when a run failure is due to a timeout
+  assert_success
+  assert_output --partial "${PROCESS}"
+
+  # Should kill the process successfully:
+  # (which should then get restarted by supervisord)
+  _run_in_container pkill --echo "${PROCESS}"
+  assert_output --partial "${PROCESS}"
+  assert_success
+
+  # Wait until original process is not running:
+  # (Ignore restarted process by filtering with MIN_PROCESS_AGE, --fatal-test with `false` stops polling on error):
+  run repeat_until_success_or_timeout --fatal-test "_check_if_process_is_running ${PROCESS} ${MIN_PROCESS_AGE}" 30 false
+  assert_output --partial "'${PROCESS}' is not running"
+  assert_failure
+
+  # Should be running:
+  # (poll as some processes a slower to restart, such as those run by wrapper scripts adding delay via sleep)
+  run_until_success_or_timeout 30 _check_if_process_is_running "${PROCESS}"
+  assert_success
+  assert_output --partial "${PROCESS}"
+}
+
+# NOTE: CONTAINER_NAME is implicit; it should have be set prior to calling.
```

Comment:
```suggestion
# NOTE: CONTAINER_NAME is implicit; it should have been set prior to calling.
```

Reply:
@georglauterbach did you want this change applied? (_I'm a little hesitant to end up in approval ping pong again_ :laughing: )